gitpod-io/gitpod
# [server] Serve IDE config via gitpod-protocol

## Description


With this PR the IDE entries on the preferences page are served from the `server-ide-configmap`. This allows the IDE team to add new IDEs by just editing the configmap.

## Related Issue(s)

Fixes #6601

## How to test

- Make sure that changing the IDE still works.
- You can test the code by changing the server-ide-configmap (e.g. change name, set `hidden`, remove entires etc.) and verifying that the changes are propagated to the dashboard. Just change the configmap, no component restart is needed.

## Release Notes

```release-note
NONE
```
